### PR TITLE
Allow migration steps to use automatic DI

### PR DESCRIPTION
**Pull request by nickvergessen**
@DeepDiver1975 @MorrisJobke @LukasReschke 

Allows migration steps to have dependencies in the constructors, like we already have for background jobs.
E.g. https://github.com/owncloud/workflow/pull/102

### CODE REVIEW

File: lib/private/Repair.php [hunk 1]
```diff
@@ -93,15 +93,20 @@ public function run() {
 	 */
 	public function addStep($repairStep) {
 		if (is_string($repairStep)) {
-			if (class_exists($repairStep)) {
-				$s = new $repairStep();
-				if ($s instanceof IRepairStep) {
-					$this->repairSteps[] = $s;
+			try {
+				$s = \OC::$server->query($repairStep);
```
Comment (DeepDiver1975) [inline lib/private/Repair.php]: but this will no load steps living in apps - right?

Comment (nickvergessen) [inline lib/private/Repair.php]: It should load them, because we look in app containers for OCA classes.
And it also worked with my workflow PR

**Comment by nickvergessen** [comment]
Bump, please review @LukasReschke @MorrisJobke @DeepDiver1975 @rullzer

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by rullzer** [comment]
:+1:

**Pull request merged**